server/source/serverlib/game/game.cpp

```diff
@@ -105,6 +115,45 @@ void Game::doCurrentPlayerBuyField()
     m_board.changeOwner(propertyId, currentPlayer().name());
 }
 
+void Game::doCurrentPlayerBuyHouse(int fieldID, bool buy)
+{
+    auto buyPrice = m_board.housePrice(fieldID);
+    auto sellPrice = buyPrice / 2;
+
+    m_board.ensureFullGroupOwnership(currentPlayer().name(), fieldID);
+
+    if (buy)
+    {
+        if (m_bank.balance(currentPlayer().name()) < buyPrice)
+        { // throw exception indirectly
+            m_bank.takeMoney(currentPlayer().name(), buyPrice);
+        }
+        m_board.buildHouse(fieldID);
+        m_bank.takeMoney(currentPlayer().name(), buyPrice);
+    }
+    else
+    {
+        m_board.removeHouse(fieldID);
+        m_bank.giveMoney(currentPlayer().name(), sellPrice);
+    }
+}
+
+void Game::doCurrentPlayerChangeHouses(const std::vector<std::pair<int,int>>& newLevels)
```

Comment:
Why do we need a buy house and change houses? Wouldn't be change houses do all the job?